pkg/operator/controller/certificate/ca.go

```diff
@@ -125,6 +126,9 @@ func desiredRouterCASecret(namespace string) (*corev1.Secret, error) {
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      name.Name,
 			Namespace: name.Namespace,
+			Annotations: map[string]string{
+				annotations.OpenShiftComponent: "Networking / router",
+			},
```

Comment:
You'll need to add update logic to `ensureRouterCASecret` if you want the operator to add this annotation on upgrade.

Reply:
I don't think its required, `if !reflect.DeepEqual` in `ensureRouterCASecret` would start an update if annotations mismatch